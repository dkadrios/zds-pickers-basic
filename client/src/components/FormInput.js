import React from 'react'
import PropTypes from 'prop-types'
import TextField from '@material-ui/core/TextField'
import withStyles from '@material-ui/core/styles/withStyles'

// istanbul ignore next
const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit * 2,
    marginRight: theme.spacing.unit,
    width: 200,
  },
})

export const FormInput = (props) => {
  const {
    classes,
    input,
    label,
    meta: { touched, error, warning },
    type,
  } = props
  const errored = touched && (error || warning)

  return (
    <TextField
      {...input}
      className={classes.textField}
      error={!!errored}
      helperText={touched ? error || warning : null}
      label={label}
      type={type}
    />
  )
}
FormInput.propTypes = {
  input: PropTypes.object.isRequired, // eslint-disable-line
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
    warning: PropTypes.string,
  }).isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(FormInput)
