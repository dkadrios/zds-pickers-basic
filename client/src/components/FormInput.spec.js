import flow from 'lodash/fp/flow'
import set from 'lodash/fp/set'
import { FormInput } from './FormInput'

const onChange = jest.fn()
const onBlur = jest.fn()

const input = { value: '', onChange, onBlur }
const meta = { touched: false, error: '', warning: '' }

const props = {
  input,
  meta,
  label: '',
  type: 'input',
  classes: {},
}

describe('FormInput tests', () => {
  it('renders correctly', () => {
    shallowExpect(FormInput)(props).toMatchSnapshot()
  })

  it('should render an error when touched', () => {
    const localProps = flow(
      set('meta.error', 'error'),
      set('meta.touched', true),
    )(props)
    shallowExpect(FormInput)(localProps).toMatchSnapshot()
  })

  it('should render a warning when touched', () => {
    const localProps = flow(
      set('meta.warning', 'warning'),
      set('meta.touched', true),
    )(props)
    shallowExpect(FormInput)(localProps).toMatchSnapshot()
  })
})
