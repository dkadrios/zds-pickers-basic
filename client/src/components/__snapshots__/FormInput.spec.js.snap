// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput tests renders correctly 1`] = `
<TextField
  error={false}
  helperText={null}
  label=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  required={false}
  select={false}
  type="input"
  value=""
  variant="standard"
/>
`;

exports[`FormInput tests should render a warning when touched 1`] = `
<TextField
  error={true}
  helperText="warning"
  label=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  required={false}
  select={false}
  type="input"
  value=""
  variant="standard"
/>
`;

exports[`FormInput tests should render an error when touched 1`] = `
<TextField
  error={true}
  helperText="error"
  label=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  required={false}
  select={false}
  type="input"
  value=""
  variant="standard"
/>
`;
